src/memory_target_ops.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#ifndef UMF_MEMORY_TARGET_OPS_H
+#define UMF_MEMORY_TARGET_OPS_H 1
+
+#include <stdbool.h>
+
+#include <umf/base.h>
+#include <umf/memspace.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct umf_memory_target_t *umf_memory_target_handle_t;
+
+typedef struct umf_memory_target_ops_t {
+    /// Version of the ops structure.
+    /// Should be initialized using UMF_VERSION_CURRENT
+    uint32_t version;
+
+    umf_result_t (*initialize)(void *params, void **memoryTarget);
+    void (*finalize)(void *memoryTarget);
+
+    umf_result_t (*pool_create_from_memspace)(
+        umf_memspace_handle_t memspace, void **memoryTargets, size_t numTargets,
+        umf_memspace_policy_handle_t policy, umf_memory_pool_handle_t *pool);
+
+    umf_result_t (*memory_provider_create_from_memspace)(
+        umf_memspace_handle_t memspace, void **memoryTargets, size_t numTargets,
```

Comment:
what if `memspace` consists of multiple `memory_targets`? How we will choose  which `memory_target` should we ask to create a pool or provider from `memspace`?

Reply:
That's why we only support memory targets of the same type for now. See here: https://github.com/oneapi-src/unified-memory-framework/blob/9c4e3dacc99ec0a303f35357eed0c18acd41ad61/src/memspace.c#L68

Reply:
I understand that today we support only memory targets of the same type (same ops structure). My question is mostly about the architecture/design, not the current implementation. if we are creating pool/provider from memspace perhaps the current design is wrong and memspace should be responsible for pool/provider creation? In the future I can imagine memspace which consists of GPU memory target and numa memory target - in that case I expect that we will choose GPU memory provider (L0 or cuda) and create USM shared pool.

What do you think?